tests/unit/core/server/testServer.js

```diff
@@ -157,20 +157,35 @@ describe(__filename, () => {
         .set('cookie', `${defaultConfig.get('cookieName')}="${token}"`)
         .end();
 
-      const { api, user } = store.getState();
+      const {
+        api,
+        user,
+        reduxAsyncConnect: reduxAsyncConnectState,
+      } = store.getState();
 
       expect(response.statusCode).toEqual(200);
       expect(api.token).toEqual(token);
       expect(user.id).toEqual(42);
       expect(user.username).toEqual('babar');
       mockUserApi.verify();
+
+      const { loadState } = reduxAsyncConnectState;
+      expect(loadState).toEqual({});
     });
 
     it('returns a 500 error page when retrieving the user profile fails', async () => {
+      const errorMessage = 'example of an API error';
+      const error = Promise.reject(new Error(errorMessage));
+      // Having `.catch()` mutates `error` internal state and avoids the Node
+      // unhandled rejection warning that Jest 21.0+ converts into an error. We
+      // do want to throw exceptions on the server. See also:
+      // http://thecodebarbarian.com/unhandled-promise-rejections-in-node.js.html
+      error.catch(() => {});
```

Comment:
This feels pretty naughty, like something that should be labelled with `HACK:` haha.

But the reasoning seems fine to me. I honestly forget *why* we want to throw on the server though, maybe you could add a comment explaining why?